rancherdesktop/rule_test.go

```diff
@@ -0,0 +1,264 @@
+package rancherdesktop
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+func newRule(appVersion, platform, arch, platformVersion, constraintsVersion string) Rule {
+	parsedAppVersion, err := semver.NewConstraint(appVersion)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse appVersion %q: %s", appVersion, err))
+	}
+	parsedPlatformVersion, err := semver.NewConstraint(platformVersion)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse platformVersion %q: %s", platformVersion, err))
+	}
+	parsedConstraintsVersion, err := semver.NewConstraint(constraintsVersion)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse constraintsVersion %q: %s", constraintsVersion, err))
+	}
+	return Rule{
+		Criteria: Criteria{
+			AppVersion:      parsedAppVersion,
+			Platform:        platform,
+			Arch:            arch,
+			PlatformVersion: parsedPlatformVersion,
+		},
+		Constraints: Constraints{
+			Version: parsedConstraintsVersion,
+		},
+	}
+}
+
+func newInstanceInfo(appVersion, platform, arch, platformVersion string) InstanceInfo {
+	parsedAppVersion, err := semver.NewVersion(appVersion)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse appVersion %q: %s", appVersion, err))
+	}
+	if platform == "" {
+		panic("must specify platform")
+	}
+	if arch == "" {
+		panic("must specify arch")
+	}
+	parsedPlatformVersion, err := semver.NewVersion(platformVersion)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse platformVersion %q: %s", platformVersion, err))
+	}
+	return InstanceInfo{
+		AppVersion:      parsedAppVersion,
+		Platform:        platform,
+		Arch:            arch,
+		PlatformVersion: parsedPlatformVersion,
+	}
+}
+
+func TestRule(t *testing.T) {
+
+	t.Run(".Validate", func(t *testing.T) {
```

Comment:
`.Validate` is the method of `Rule` that we're testing here. We do this in our jest tests so I figured it would make sense, but Go's test output doesn't make it as clear. I could change it to `.Validate method` I guess? Let me know what you think.